### PR TITLE
UI: Fix toggle button for custom theming

### PR DESCRIPTION
**Pull request by naz3eh**
Issue: [UI: Fix toggle button for custom theming #13924](https://github.com/storybookjs/storybook/issues/13924)

**Fixes Boolean control change, Search input theme configuration, and arrows in the left nav.**

## What I did
Forked the storybook repo, and setup locally as specified [here](https://storybook.js.org/docs/react/contribute/code). 
When I was going through the code, I noticed that Control was taking the background parameters from **barBg**, the change will take the parameters from **app**.

## How to test

Run ```yarn test``` in storybook and all test suites and tests should pass.
image

- Is this testable with Jest or Chromatic screenshots?
  I am not sure, as I am new and contributing first time to storybook. 😅 
- Does this need a new example in the kitchen sink apps?
  No, I don't think it requires. But, I am not sure.
- Does this need an update to the documentation?
  No, according to me, it doesn't require an update to the documentation.

### CODE REVIEW

File: lib/ui/src/components/sidebar/Tree.tsx [hunk 1]
```diff
@@ -39,10 +39,10 @@ export const Action = styled.button(({ theme }) => ({
       : transparentize(0.6, theme.color.defaultText),
 
   '&:hover': {
-    color: theme.barSelectedColor,
+    color: theme.background.app,
```
Comment (darleendenno) [inline lib/ui/src/components/sidebar/Tree.tsx:42]:
```suggestion
    color: theme.color.secondary,
```

File: lib/ui/src/components/sidebar/Tree.tsx [hunk 2]
```diff
@@ -39,10 +39,10 @@ export const Action = styled.button(({ theme }) => ({
...
   },
   '&:focus': {
-    color: theme.barSelectedColor,
+    color: theme.background.app,
```
Comment (darleendenno) [inline lib/ui/src/components/sidebar/Tree.tsx:45]:
```suggestion
    color: theme.color.secondary,
```

File: lib/components/src/controls/Boolean.tsx [hunk 3]
```diff
@@ -70,8 +70,8 @@ const Label = styled.label(({ theme }) => ({
     },
   },
 
-  'input:checked ~ span:last-of-type, input:not(:checked) ~ span:first-of-type': {
-    background: theme.background.bar,
+  'input:checked ~ span:first-of-type, input:not(:checked) ~ span:last-of-type': {
+    background: theme.background.app,
     boxShadow: `${opacify(0.1, theme.appBorderColor)} 0 0 2px`,
     color: theme.color.defaultText,
     padding: '7px 15px',
```
Comment (darleendenno) [inline lib/components/src/controls/Boolean.tsx:77]: Let's chat with design about this change -- what is the expected text/background style of these toggle switches @MichaelArestad?

I'm leaning towards this implementation:
```  'input:checked ~ span:first-of-type, input:not(:checked) ~ span:last-of-type': {
    background: theme.background.bar,
    boxShadow: `${opacify(0.1, theme.appBorderColor)} 0 0 2px`,
    color: theme.color.inverseText,
    padding: '7px 15px',
```

when tested with the default custom theme, it looks like this:
image

**Review by darleendenno — CHANGES_REQUESTED** [review]
Hi @Nazeeh21 👋 Thank you so much for contributing to Storybook and fixing a portion of our custom theming. We really appreciate your help 🎉 

I noticed the sidebar Tree sub-navigation icons are still disappearing on hover, so let's address that and get some input from designers before we merge.

I'll share with you how you can test out your theme locally to ensure it's exactly what you expect. This article will show you how to set up your [default custom theme](https://storybooks.netlify.app/configurations/theming/) -- I recommend adding this custom theme to `examples/official-storybook` so you can see it when you run `yarn start`.

Before running `yarn start`, run `yarn build` and select the `components` package. This will ensure your theme is built and displays in the browser. I hope this helps! Please reach out if you run into any issues; I'm happy to help ❤️

**Comment by Nazeeh21** [comment]
Hey @darleendenno! Thanks for the suggestion! 
I will surely implement the suggested changes soon.

**Comment by Nazeeh21** [comment]
I have implemented the suggested changes. Kindly review it and please provide me suggestions and changes if any. Thanks! 😃

File: lib/components/src/controls/Boolean.tsx (hunk 3 above)
Comment (MichaelArestad) [reply lib/components/src/controls/Boolean.tsx:77]: > what is the expected text/background style of these toggle switches

Good question. @domyen would know better than me. I'm not super familiar with Storybook theming yet.

What I might expect is that the text and background styles of the unselected boolean control is:
* The active item's background color could match the theme's background color .
* The active item's text color should be higher contrast against it's background than the inactive item's color/background contrast.

For example:
image

Comment (darleendenno) [reply lib/components/src/controls/Boolean.tsx:77]: Thanks @MichaelArestad -- @Nazeeh21 implemented this beautifully 🚀

**Review by darleendenno — APPROVED** [review]
Thank you @Nazeeh21 🚀 🎸 🎉  I've looked at these components locally with several custom themes -- The search bar, boolean and side bar icons look **SO** much better thanks to your work ❤️ 
```
                            
            🌕 
 
                                  

🚀 
```
@shilman for final approval & merge

**Review by shilman — APPROVED** [review]
thanks @Nazeeh21 ! 🚀

**Pull request merged**